### PR TITLE
Issue #183 - add source level debug to mpConfig

### LINKED ISSUE #183: mpConfig: Add further debug and serviceablity to the config impl
Need to provide a way to figure out which config property is from which config source.
We need to add debug info when config.getValue or getOptionalValue is called.

### PR DESCRIPTION
**Pull request by tevans78**
fixes #183 
#build

### CODE REVIEW

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java [hunk 1]
```diff
@@ -116,21 +114,41 @@ public T get() {
...
+
+        return compositeValue;
     }
 
     /**
-     * Select a non null preferably from two values
+     * Fetch the latest version of the property. If not up to date then resolve to the latest
+     * value, inline.
+     *
+     * TODO: Make resolving property value an offline task
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java]: We should not cargo copy this TODO over into our method as we do not want to DO that.

File: dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java [hunk 2]
```diff
@@ -139,14 +157,14 @@ private T firstNonNull(T first, T second) {
      * @return
      * @throws Exception
      */
-    protected abstract T resolveCurrent() throws Exception;
+    protected abstract CachedCompositeValue<T> resolveCurrent() throws Exception;
 
     /** {@inheritDoc} */
     @Override
     public int hashCode() {
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java]: Needs a javadoc comment as so odd and looks like a missing bit (the value) but probably as intended (will be odd though if used as a key in a map? as all Strings will have the same hash).

**Comment by tevans78** [comment]
New issue opened to capture the suggested improvements... https://github.com/OpenLiberty/open-liberty/issues/262

**Comment by tevans78** [comment]
#build

**Pull request merged**